pwiz_tools/Shared/PanoramaClient/PanoramaUtil.cs

```diff
@@ -236,88 +120,67 @@ private static bool IsValidEnsureLoginResponse(JObject jsonResponse, string expe
-                return new UserState(UserStateEnum.unknown, e.Message, GetEnsureLoginUri(pServer));
-            }
-        }
-
-        public static void VerifyFolder(IPanoramaClient panoramaClient, PanoramaServer server, string panoramaFolder)
-        {
-            switch (panoramaClient.IsValidFolder(panoramaFolder, server.Username, server.Password))
-            {
-                case FolderState.notfound:
-                    throw new PanoramaServerException(
-                        string.Format(
-                            Resources.PanoramaUtil_VerifyFolder_Folder__0__does_not_exist_on_the_Panorama_server__1_,
-                            panoramaFolder, panoramaClient.ServerUri));
-                case FolderState.nopermission:
-                    throw new PanoramaServerException(string.Format(
-                        Resources
-                            .PanoramaUtil_VerifyFolder_User__0__does_not_have_permissions_to_upload_to_the_Panorama_folder__1_,
-                        server.Username, panoramaFolder));
-                case FolderState.notpanorama:
-                    throw new PanoramaServerException(string.Format(
-                        Resources.PanoramaUtil_VerifyFolder__0__is_not_a_Panorama_folder,
-                        panoramaFolder));
-            }
-        }
-
         /// <summary>
-        /// Parses the JSON returned from the getContainers LabKey API to look for the folder type and active modules in a container.
+        /// Parses the JSON returned from the getContainers LabKey API to look for the folder type for a container.
         /// </summary>
         /// <param name="folderJson"></param>
-        /// <returns>True if the folder is a Targeted MS folder.</returns>
-        public static bool CheckFolderType(JToken folderJson)
+        /// <returns>True if the folder type is "Targeted MS".</returns>
+        public static bool IsTargetedMsFolder(JToken folderJson)
         {
             if (folderJson != null)
             {
-
                 var folderType = (string)folderJson[@"folderType"];
-                var modules = folderJson[@"activeModules"];
-                return modules != null && ContainsTargetedMSModule(modules) &&
-                       Equals(@"Targeted MS", folderType);
+                return Equals(@"Targeted MS", folderType); // A "Targeted MS" type folder will always have the "TargetedMS" module enabled.
             }
 
             return false;
         }
 
-        /// <summary>
-        /// Parses the JSON returned from the getContainers LabKey API to look user permissions in the container.
-        /// </summary>
-        /// <param name="folderJson"></param>
-        /// <returns>True if the user has insert permissions.</returns>
-        public static bool CheckFolderPermissions(JToken folderJson)
+        public static bool HasTargetedMsModule(JToken folderJson)

```

Comment:
Consider using just one function for determining targeted MS nature of a folder.